R/drive_read.R

```diff
@@ -0,0 +1,298 @@
+#'   destfile = tfile
+#' )
+#' imdb_latin1 <- tfile %>%
+#'   drive_upload(name = "imdb_latin1.csv")
+#' imdb_latin1 %>%
+#'   drive_read(encoding = "ISO-8859-1") %>% # "latin1" works too
+#'   read.csv(text = .)
+#' if (require(readr)) {
+#'   imdb_latin1 %>%
+#'     drive_read(encoding = "latin1") %>%  # "ISO-8859-1" works too
+#'     readr::read_csv()
+#' }
+#'
+#' # clean up comma-separated values with latin-1 encoding
+#' drive_rm(imdb_latin1)
+#'
+#' # Google Doc --> character vector
+#' chicken_doc <- drive_example("chicken.txt") %>%
+#'   drive_upload(type = "document")
+#' chicken_doc %>%
+#'   # NOTE: we must specify an export MIME type
+#'   drive_read(type = "text/plain") %>%
+#'   strsplit(split = "(\r\n|\r|\n)") %>%
+#'   .[[1]]
+#'
+#' # clean up Google Doc
+#' drive_rm(chicken_doc)
+#'
+#' # JPEG --> raw --> raster
+#' # https://stat.ethz.ch/R-manual/R-patched/doc/html/logo.jpg
+#' if (require(jpeg)) {
+#'   r_logo <- R.home("doc/html/logo.jpg") %>%
+#'      drive_upload(name = "r_logo.jpg")
+#'   img <- r_logo %>%
+#'     drive_read() %>%
+#'     jpeg::readJPEG()
+#'   plot(0:1, 0:1, type = "n")
+#'   rasterImage(img, 0, 0, 1, 1)
+#'
+#'   # JPEG cleanup
+#'   drive_rm(r_logo)
+#' }
+#'
+#' # html -> character vector
+#' # https://stat.ethz.ch/R-manual/R-patched/doc/html/about.html
+#' if (require(rvest) && require(readr)) {
+#'   r_about <- R.home("doc/html/about.html") %>%
+#'      drive_upload()
+#'   r_about %>%
+#'     drive_read() %>%
+#'     rvest::html_text2() %>%
+#'     readr::read_lines() %>%
+#'     lapply(substr, start = 1, stop = 70)
+#'
+#'   # html clean up
+#'   drive_rm(r_about)
+#' }
+#'
+#' # html --> character vector
+#' if (require(rvest) && require(stringi)) {
+#'   tfile <- tempfile(fileext = ".html")
+#'   curl::curl_download("https://httpbin.org/html", destfile = tfile)
+#'   httpbin_html <- tfile %>%
+#'     drive_upload(name = "httpbin.html")
+#'   httpbin_html %>%
+#'     drive_read() %>%
+#'     rvest::html_text2() %>%
+#'     stringi::stri_split_boundaries(type = "sentence") %>%
+#'     lapply(substr, start = 1, stop = 70) %>%
+#'     .[[1]] %>%
+#'     head(6)
+#'
+#'   # html clean up
+#'   drive_rm(httpbin_html)
+#' }
+#'
+#' # xml --> list
+#' if (require(xml2)) {
+#'   tfile <- tempfile(fileext = ".xml")
+#'   curl::curl_download("https://httpbin.org/xml", destfile = tfile)
+#'   httpbin_xml <- tfile %>%
+#'     drive_upload()
+#'   httpbin_xml %>%
+#'     drive_read() %>%
+#'     xml2::as_list()
+#'
+#'   # xml clean up
+#'   drive_rm(httpbin_xml)
+#' }
+#'
+#' # json --> list
+#' tfile <- tempfile(fileext = ".json")
+#' curl::curl_download("https://httpbin.org/json", destfile = tfile)
+#' httpbin_json <- tfile %>%
+#'   drive_upload()
+#' httpbin_json %>%
+#'   drive_read()
+#'
+#' # json clean up
+#' drive_rm(httpbin_json)
+drive_read <- function(file,
+                       type = NULL,
+                       as = c("auto", "string", "raw", "json", "html", "xml"),
```

Comment:
I am exposing the functionality of the `httr2::resp_body_*()` family of functions, but, for now, implementing with httr(not 2):

* `resp_body_raw()` returns the raw bytes.
* `resp_body_string()` returns a UTF-8 string.
* `resp_body_json()` returns parsed JSON.
* `resp_body_html()` returns parsed HTML.
* `resp_body_xml()` returns parsed XML.

https://httr2.r-lib.org/reference/resp_body_raw.html